### ISSUE TITLE
Feature Request: support saving to s3 (or s3 compatible)

### ISSUE THREAD

**Issue by fifofefe** [gildas-lormeau/SingleFile#1487]
**Is your feature request related to a problem? Please describe.**
No, ideally it would just be another entry in the "Destination" menu.

**Describe the solution you'd like**
I would like to upload the html file to a S3 bucket or other object storages. Ideally I would give to the extension a URL, a bucket name, the necessary auth tokens and the extension would upload the file to the bucket.

**Additional context (optional)**
I do not have direct experience in this type of functionality, so I am not sure whether this is a plausible feature to implement...

**Comment by gildas-lormeau** [comment]
I agree this would be useful. From a technical point of view, it's not as simple as the other destinations, but it can be done. This is because SingleFile would have to send a "presigned request". 

For the record, this is documented here: https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-post-example.html. There are also a few (pure) JavaScript implementations, for example:
- https://github.com/odysseyscience/react-s3-uploader/blob/master/s3upload.js
- https://github.com/asafdav/ng-s3upload/blob/master/src/ng-s3upload/services/s3-uploader.js

**Comment by gildas-lormeau** [comment]
The new destination "upload to S3" is implemented. It will be available in the next version.
